src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/RejectingExecutor.java

```diff
@@ -0,0 +1,84 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * An executor which throws an {@link RejectedExecutionException} any time it is asked to execute a task
+ * This class is useful for testing certain branches of our net code
+ */
+class RejectingExecutor implements ExecutorService {
```

Comment:
This is good but I see that it's being used by test classes (and very rightly so). So, should we move it to the test packing instead of keeping it here ?

Reply:
It is already under src/test